pkg/cloudprovider/instance.go

```diff
@@ -304,23 +304,16 @@ func (p *InstanceProvider) getInstance(ctx context.Context, id string) (*ec2.Ins
 	if *instance.State.Name == ec2.InstanceStateNameTerminated {
 		return nil, awserrors.InstanceTerminatedError{Err: fmt.Errorf("instance is in terminated state")}
 	}
-	if awssettings.FromContext(ctx).NodeNameConvention == awssettings.ResourceName {
-		return instance, nil
-	}
 	if len(aws.StringValue(instance.PrivateDnsName)) == 0 {
 		return nil, multierr.Append(err, fmt.Errorf("got instance %s but PrivateDnsName was not set", aws.StringValue(instance.InstanceId)))
 	}
 	return instance, nil
 }
 
-func (p *InstanceProvider) instanceToNode(ctx context.Context, instance *ec2.Instance, instanceTypes []cloudprovider.InstanceType) *v1.Node {
+func (p *InstanceProvider) instanceToNode(instance *ec2.Instance, instanceTypes []cloudprovider.InstanceType) *v1.Node {
 	for _, instanceType := range instanceTypes {
 		if instanceType.Name() == aws.StringValue(instance.InstanceType) {
 			nodeName := strings.ToLower(aws.StringValue(instance.PrivateDnsName))
-			if awssettings.FromContext(ctx).NodeNameConvention == awssettings.ResourceName {
```

Comment:
I'm confused -- how does this fix the problem? This should use the resource name instead of the dns name
```
i-00a510b73a20d7d5f.us-west-2.compute.internal       Ready      <none>   4h32m   v1.23.12-eks-a64d4ad
```
Essentially, it's `${resourceName}.${region}.compute.internal`

Reply:
Maybe I'm confused, but AFAIK the resource based naming still uses the dns name. The only difference is that the dns name has the instance id instead of
The hyphenated private ip address? Do you have links to docs about using just the instance id?

Reply:
Oo I think you misread the logs up top. The top node is the new one I launched in a subnet with RBN enabled. The other two nodes are using non-RBN node names since this was an existing cluster.

Reply:
Ah gotcha.